From 70ec097851642885d650b2f6aea9aa9236874255 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:10:02 -0400
Subject: [usb] improve "not suspended yet" message in hub_suspend()

Message-id: <1410887463-20674-531-git-send-email-dzickus@redhat.com>
Patchwork-id: 95390
O-Subject: [RHEL7 PATCH 530/591] usb: improve "not suspended yet" message in hub_suspend()
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit b658b8f5c09d3d1db9641c9d520b36b3e4c0d744
Author: Dan Williams <dan.j.williams@intel.com>
Date:   Tue Jun 17 16:16:22 2014 -0700

    usb: improve "not suspended yet" message in hub_suspend()

    Reading through a recent bug report [1], Alan notes:

      "Dan, the warning message in hub_suspend() should mention that the
       child device isn't suspended yet."

    ...update the warning from:

      "usb usb3-port4: not suspended yet"

    ...to:

       "usb usb3-port4: device 3-4: not suspended yet"

    [1]: http://marc.info/?l=linux-usb&m=140290586301336&w=2

    Reported-by: Alan Stern <stern@rowland.harvard.edu>
    Signed-off-by: Dan Williams <dan.j.williams@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/core/hub.c b/drivers/usb/core/hub.c
index 8c5393d..4a253c5 100644
--- a/drivers/usb/core/hub.c
+++ b/drivers/usb/core/hub.c
@@ -3452,7 +3452,8 @@ static int hub_suspend(struct usb_interface *intf, pm_message_t msg)
   struct usb_device *udev = port_dev->child;
 
   if (udev && udev->can_submit) {
-   dev_warn(&port_dev->dev, "not suspended yet\n");
+   dev_warn(&port_dev->dev, "device %s not suspended yet\n",
+     dev_name(&udev->dev));
    if (PMSG_IS_AUTO(msg))
     return -EBUSY;
   }
-- 
1.7.1